.github/workflows/ci.yml

```
@@ -128,7 +128,7 @@ jobs:
   docker-alpine:
     if: ${{ inputs.tag != '' || (github.ref != format('refs/heads/{0}', github.event.repository.default_branch) && !contains(github.ref, 'refs/pull')) }}
     name: Docker Alpine (Binary)
-    uses: stacks-network/stacks-blockchain/.github/workflows/image-build-alpine-binary.yml@master
+    uses: jbencin/stacks-blockchain/.github/workflows/image-build-alpine-binary.yml@ci/pass-without-dockerhub-credentials
```

Should this just be:

```suggestion
    uses: ./.github/workflows/image-build-alpine-binary.yml
```

That way, it always uses the workflow from the current branch? @wileyj

---

I agree with this and will modify the PR to include it. I didn't understand the reason for `@master` (it complicates CI testing IMO), but I didn't want to change too much and complicate the PR.

---

hmm, this is actually something i've run into in the past. 
for testing, i definitely use the filepath `./.github/...` but there was a decision made long ago to only use the action in the master branch :
```
    uses: stacks-network/stacks-blockchain/.github/workflows/image-build-alpine-binary.yml@master
```
that way, the action that runs will always be a PR approved workflow vs some workflow in a branch. 

unless there's a reason to keep the filepath that i'm not seeing, i'd prefer to keep the standard we've been using for a while of requiring actions to run off of the master branch.

---

> that way, the action that runs will always be a PR approved workflow vs some workflow in a branch.

That seems pretty easy to get around though, since anyone that wants to run their own unapproved workflow can do the `./.github...` trick also, right?

I think a better solution for preventing unapproved workflows being run (which does matter here because CI has DockerHub secrets) is to restrict write access to the repo. Instead, contributors could fork it to their own namespace and issue PRs from there (which is what I do and why I'm making this change)

---

yea, i see  your point and i don't entirely disagree with you. in my fork, i also have to make similar changes to test things, so i'm acutely aware of the problem you're trying to solve. 
i just think this PR is the wrong place to push for that change. 

i think it would be better suited to have another PR where we can answer that question of what standard(s) do we want to use for localized workflows. 
I know @kantai has opinions on workflow versioning (in context, he had raised concerns about using un-versioned external workflows so it's not entirely the same with a localized workflow), and if we're going to change which workflows we allow to run, i'd like to have more input from others. 

incidentally, i do have an open PR to modify a lot of the workflows. perhaps that's a good place to have this discussion about resuable workflows? https://github.com/stacks-network/stacks-blockchain/pull/3847

---

@jbencin would you also be able to port your changes to the branch i'm trying to PR in #3487 ? if not, i can copypasta

---

@wileyj Sure I'll take a look at that today. You're referring to the changes with `DOCKER_PUSH` I assume, and not `./.github...`?

---

I made both changes in separate commits. See PR #3895